components/telemetry-operator/controller/tracepipeline/setup.go

```diff
@@ -29,6 +29,11 @@ func (r *Reconciler) SetupWithManager(mgr ctrl.Manager) error {
 			&source.Kind{Type: &corev1.Secret{}},
 			handler.EnqueueRequestsFromMapFunc(r.mapSecret),
 			builder.WithPredicates(onlyUpdate()),
+		).
+		Watches(
```

Comment:
This should be unnecessary. The owner reference should already trigger a reconciliation if the Deployment changes.

Reply:
this code slices removed.